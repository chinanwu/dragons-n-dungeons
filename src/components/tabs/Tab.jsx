import PropTypes from 'prop-types';
import React from 'react';

// This is basically a shell, I wanted this component to be used as an intuitive way to make tabs
const Tab = ({ name, children, disabled }) => {
  return <div />;
};

Tab.propTypes = {
  name: PropTypes.string,
  children: PropTypes.any,
  disabled: PropTypes.bool,
};

Tab.defaultProps = {
  disabled: false,
};

export default Tab;
