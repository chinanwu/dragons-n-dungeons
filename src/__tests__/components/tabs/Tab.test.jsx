import { shallow } from 'enzyme';
import React from 'react';
import Tab from '../../../components/tabs/Tab.jsx';

jest.unmock('../../../components/tabs/Tab.jsx');

describe('Tab component', () => {
  describe('rendering', () => {
    it('renders Tab', () => {
      const wrapper = shallow(<Tab />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
