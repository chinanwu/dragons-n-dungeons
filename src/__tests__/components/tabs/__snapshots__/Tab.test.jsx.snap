// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component rendering renders Tab 1`] = `<div />`;
